gtsfm/scene_optimizer.py

```diff
@@ -18,31 +18,34 @@
 from gtsam import (
     Pose3,
     SfmData,
-    Unit3,
 )
 
 import gtsfm.utils.geometry_comparisons as comp_utils
 import gtsfm.utils.io as io_utils
 import gtsfm.utils.logger as logger_utils
 import gtsfm.utils.serialization  # import needed to register serialization fns
 import gtsfm.utils.viz as viz_utils
-from gtsfm.averaging.rotation.rotation_averaging_base import (
-    RotationAveragingBase,
-)
-from gtsfm.averaging.translation.translation_averaging_base import (
-    TranslationAveragingBase,
-)
 from gtsfm.common.image import Image
 from gtsfm.common.keypoints import Keypoints
 from gtsfm.feature_extractor import FeatureExtractor
-from gtsfm.frontend.detector_descriptor.detector_descriptor_base import (
-    DetectorDescriptorBase,
-)
-from gtsfm.frontend.matcher.matcher_base import MatcherBase
-from gtsfm.frontend.verifier.verifier_base import VerifierBase
 from gtsfm.multi_view_optimizer import MultiViewOptimizer
 from gtsfm.two_view_estimator import TwoViewEstimator
 
+# paths for storage
+PLOT_PATH = "plots"
+PLOT_CORRESPONDENCE_PATH = os.path.join(PLOT_PATH, "correspondences")
+METRICS_PATH = "result_metrics"
+RESULTS_PATH = "results"
+
+"""
+data type for frontend metrics on a pair of images, containing:
+1. rotation angular error
+2. translation angular error
+3. number of correct correspondences
+4. inlier ratio
+"""
+FRONTEND_METRICS_FOR_PAIR = Tuple[Optional[float], Optional[float], int, float]
```

Comment:
why are only the first two optional @ayushbaid ? not all 4?

Reply:
When the verification fails, the number of correct correspondences is zero and inlier ratio is 0 too. Hence I don't need optional for them.